returnn/tf/layers/basic.py

```diff
@@ -913,37 +913,199 @@ def transform_config_dict(cls, d, network, get_layer):
+      input_data.placeholder,
+      common_axes_input + input_axes[:gather_axis-batch_dims] + [old_gather_axis] + input_axes[gather_axis-batch_dims:])
+    # (BatchAxes.., PositionAxes..)
+    indices = tf.transpose(position_data.placeholder, common_axes_position + position_axes)
+    # (BatchAxes.., PositionAxes.., InputAxes..)
+    self.output.placeholder = tf.gather(params=params, indices=indices, axis=gather_axis, batch_dims=batch_dims)
+
+    # Take size_placeholder from input_data if they exist there, otherwise from position_data
+    size_placeholder = dict()
+    for input_axis, size in input_data.size_placeholder.items():
+      input_axis = input_data.get_batch_axis(input_axis)
+      if input_axis == old_gather_axis:
+        continue
+      output_axis = self.output.get_batch_axis_excluding_batch(
+        self.translate_input_axis(input_axis, old_gather_axis, *axes))
+      size_placeholder[output_axis] = size
+    for position_axis, size in position_data.size_placeholder.items():
+      position_axis = position_data.get_batch_axis(position_axis)
+      output_axis = self.output.get_batch_axis_excluding_batch(self.translate_position_axis(position_axis, *axes))
+      size_placeholder.setdefault(output_axis, size)
+    self.output.size_placeholder = size_placeholder
+
+  @classmethod
+  def _get_common_input_position_axes(cls, input_data, position_data, old_gather_axis):
+    """
+    Determine all common axes of input_data and position_data. All other axes will be specific axes then (except for
+    the gather axis of ``input_data``)
+    :param Data input_data:
+    :param Data position_data:
+    :param int old_gather_axis: gather axis of ``input_data`` counted with batch dim (before any transformation)
+    :rtype: (list[int], list[int], list[int], list[int])
+    :return: (common_axes_input, common_axes_position, specific_input_axes, specific_position_axes), all counted with
+    batch dim.
+    """
+    is_equal_opts = dict(ignore_feature_dim=True, allow_same_spatial_dim=True, broadcast_matches=True)
+    common_axes_pairs = [
+      (input_axis, position_axis)
+      for input_axis in range(input_data.batch_ndim) for position_axis in range(position_data.batch_ndim)
+      if not input_axis == old_gather_axis
+      if input_data.get_dim_tag(input_axis).is_equal(position_data.get_dim_tag(position_axis), **is_equal_opts)
+    ]
+    common_axes_input, common_axes_position = zip(*common_axes_pairs) if common_axes_pairs else ([], [])
+    common_axes_input, common_axes_position = list(common_axes_input), list(common_axes_position)
+    specific_input_axes = [
+      axis for axis in range(input_data.batch_ndim) if axis not in common_axes_input and not axis == old_gather_axis]
+    specific_position_axes = [axis for axis in range(position_data.batch_ndim) if axis not in common_axes_position]
+    return common_axes_input, common_axes_position, specific_input_axes, specific_position_axes
 
+  # noinspection PyUnusedLocal
```

Comment:
Don't use `noinspection`. Why do you need that?

Reply:
This is for the common_axes_position argument, which I just added for completeness s.t. I can call the function with `translate_input_axis(ax, gather_axis, *axes)`.
I think Pycharm gave me a warning that the argument was unused even though I had the `del common_axes_position` statement there. I can try without :D

Reply:
I don't understand. Why would you call the function with random arguments you don't need?